src/send/SendSelectRecipient.tsx

```diff
@@ -130,34 +126,123 @@ const getStartedStyles = StyleSheet.create({
-  const throttledSearch = throttle((searchInput: string) => {
-    setSearchQuery(searchInput)
-  }, SEARCH_THROTTLE_TIME)
+  const forceTokenId = route.params?.forceTokenId
+  const defaultTokenIdOverride = route.params?.defaultTokenIdOverride
 
+  const [sendOrInviteButtonHidden, setSendOrInviteButtonHidden] = useState(true)
+  const [showInviteModal, setShowInviteModal] = useState(false)
   const [showContacts, setShowContacts] = useState(false)
+  const [showSearchResults, setShowSearchResults] = useState(false)
 
-  const recentRecipients = useSelector((state) => state.send.recentRecipients)
-  const contactsCache = useSelector(phoneRecipientCacheSelector)
-  const contactRecipients = useMemo(
-    () => sortRecipients(Object.values(contactsCache)),
-    [contactsCache]
-  )
-  const showGetStarted = !recentRecipients.length
+  const onSearch = (searchQuery: string) => {
+    // Always unset the selected recipient and hide the send/invite button
+    // when the search query is changed in order to prevent edge cases
+    // where the button appears but is bound to a recipient that is
+    // not present on the page.
+    unsetSelectedRecipient()
+    setSendOrInviteButtonHidden(true)
+    setShowSearchResults(!!searchQuery)
+  }
+  const { contactRecipients, recentRecipients } = useSendRecipients()
+  const { mergedRecipients, searchQuery, setSearchQuery } = useMergedSearchRecipients(onSearch)
 
-  const { recipientVerificationStatus, recipient, setSelectedRecipient } =
+  const { recipientVerificationStatus, recipient, setSelectedRecipient, unsetSelectedRecipient } =
     useFetchRecipientVerificationStatus()
 
-  const dispatch = useDispatch()
+  const setSelectedRecipientWrapper = (selectedRecipient: Recipient) => {
+    setSelectedRecipient(selectedRecipient)
+    setSendOrInviteButtonHidden(false)
+  }
 
   const onContactsPermissionGranted = () => {
     dispatch(importContacts())
     setShowContacts(true)
   }
 
+  const shouldShowClipboard = (content: string) => {
+    return content !== searchQuery && isAddressFormat(content)
+  }
+
+  const showGetStarted = !recentRecipients.length
+
+  const sendOrInviteButtonDisabled =
+    !!recipient && recipientVerificationStatus === RecipientVerificationStatus.UNKNOWN
+  const shouldInviteRecipient =
+    !sendOrInviteButtonDisabled &&
+    recipient?.e164PhoneNumber &&
+    recipientVerificationStatus === RecipientVerificationStatus.UNVERIFIED
+  const onPressSendOrInvite = () => {
+    if (!recipient) {
+      return
+    }
+    if (shouldInviteRecipient) {
+      setShowInviteModal(true)
+    } else {
+      navigate(Screens.SendAmount, {
+        isFromScan: false,
+        defaultTokenIdOverride,
+        forceTokenId,
+        recipient,
+        origin: SendOrigin.AppSendFlow,
+      })
+    }
+  }
+  const renderSendOrInviteButton = () => {
+    if (sendOrInviteButtonHidden) {
+      return <></>
+    }
+    return (
+      <Button
```

Comment:
Padding seems off here, might need adjusting SafeAreaView edges
<img src="https://github.com/valora-inc/wallet/assets/5062591/ba1fc9aa-e96b-4671-b63f-77dc31127a69" width="250" />

Reply:
Ah, does this just need to have `bottom` in the edges list?

Reply:
Ah, added some padding to the bottom.